### ISSUE TITLE
[Bug] Image understanding and paraphrase accuracy

### ISSUE THREAD

**Issue by dezhouc2** [aws-samples/bedrock-claude-chat#435]
## Describe the bug

Sorry for the misunderstanding (not really a bug). Would it be possible to increase the number of tokens? Currently, 4096 tokens are not enough because I am trying to implement few-shot chain of thought within the prompt. Additionally, the knowledge database's paraphrasing capability is sometimes poor. It does not capture the question when an alternative form of the question is asked. See the example below.

Original question: How to change my user settings?

original answer: 

**Unfortunately, the contexts provided do not contain any information about how to change user settings on the Uptrillion Merchant Portal. The contexts cover topics such as managing users, roles, transactions, and profile settings,  but do not specifically mention how to change user settings.Without any relevant information in the given contexts, I do not have enough information to provide the steps for changing user settings on the portal. The documentation provided does not seem to cover this particular functionality.If you have any additional context or documentation that covers how to change user settings on the Uptrillion Merchant Portal, I would be happy to review that and provide a more detailed response. However, based solely on the information  provided in the current contexts, I do not have enough information to describe the steps for changing user settings.**

Alternative question: How should I check my profile?

alternative answer:

According to the contexts provided, to check your profile on the Uptrillion merchant portal, you can follow these steps:
Click the "Profile" option on the main menu.
On the Profile screen, your profile information, such as username, first name, last name, and email address, will be displayed.
The contexts state:
"Editing Profile
Step 1. Click Profile on the main menu.
Step 2. Click the Edit button.
Step 3. Edit the profile, including nick name, first name, and last name, as you want and click the Submit button.
Step 4. A message "Change profile successful." is prompted, indicating the profile has been edited successfully. Click the Ok button."
So in summary, to check your profile, you need to navigate to the "Profile" section on the main menu. Your current profile information will be displayed on this screen.
Source: [Editing Profile] - mPOS Merchant Quick Start Guide

From above, you can see that "original question" and "alternative question" are the same but do not capture by knowledge database. How to resolve question like this?

**Comment by statefb** [comment]
Now the token limit is larger than 4096. We need to modify the frontend so that the app can handle large chunks.

**Comment by dezhouc2** [comment]
> Now the token limit is larger than 4096. We need to modify the frontend so that the app can handle large chunks.

Thanks for the reply. I'm also wondering how knowledge database can resolve some issues like paraphrasing? For example, if I ask the same question in an alternative way, the response will be "Unfortunately, the contexts provided do not contain any information ... "

**Comment by statefb** [comment]
![image](https://github.com/user-attachments/assets/4f176acf-4af6-46f2-845a-52be5d4b1cec)
Now the app has limitation shown as the attached. If you need to paste all contexts to the chat, we need to modify the MAX value. Please note that this parameter is not related to RAG.

**Comment by dezhouc2** [comment]
> ![image](https://private-user-images.githubusercontent.com/23316627/348918262-4f176acf-4af6-46f2-845a-52be5d4b1cec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNDE0MzgsIm5iZiI6MTcyMTE0MTEzOCwicGF0aCI6Ii8yMzMxNjYyNy8zNDg5MTgyNjItNGYxNzZhY2YtNGFmNi00NmYyLTg0NWEtNTJiZTVkNGIxY2VjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDE0NDUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlZjg0MjRjMTIyZmQ2NWRhYTc5ZDU2NzkyODk5YzZmZGVlMGVlMzM5NGVkYjU5OGRlN2M2NTgzNTczZTcwMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CBan1m6wm2Fn7rWleOmnUvvRVjLcTBezoZXWScfucgU) Now the app has limitation shown as the attached. If you need to paste all contexts to the chat, we need to modify the MAX value. Please note that this parameter is not related to RAG.



> ![image](https://private-user-images.githubusercontent.com/23316627/348918262-4f176acf-4af6-46f2-845a-52be5d4b1cec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNDE0MzgsIm5iZiI6MTcyMTE0MTEzOCwicGF0aCI6Ii8yMzMxNjYyNy8zNDg5MTgyNjItNGYxNzZhY2YtNGFmNi00NmYyLTg0NWEtNTJiZTVkNGIxY2VjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDE0NDUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlZjg0MjRjMTIyZmQ2NWRhYTc5ZDU2NzkyODk5YzZmZGVlMGVlMzM5NGVkYjU5OGRlN2M2NTgzNTczZTcwMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CBan1m6wm2Fn7rWleOmnUvvRVjLcTBezoZXWScfucgU) Now the app has limitation shown as the attached. If you need to paste all contexts to the chat, we need to modify the MAX value. Please note that this parameter is not related to RAG.

Thanks for the reply. When can I see this? I'm still confused with how maximum token number is related to paraphrasing capability of knowledge database.

**Comment by k70suK3-k06a7ash1** [comment]
@dezhouc2 
https://github.com/aws-samples/bedrock-claude-chat/pull/449/files

I have increased the max token.

**Comment by statefb** [comment]
@k70suK3-k06a7ash1 Thank you so much!
@dezhouc2 I'd be appreciated if you could wait until the PR merged.

**Comment by statefb** [comment]
@dezhouc2 
Sorry, I misunderstood. The concepts of context window and max tokens are different. The context window (currently 200K) pertains to the input to the LLM, while max tokens (the 4096 you mentioned) is a parameter related to the output. The context window is set automatically, so considering that the current context window size of 200K is sufficient for most issues, your problem likely arises from not being captured in the search results. Please try increasing the number of search results and try again. If it still doesn't work, you will need to check the cited search results in the CloudWatch logs.
